python/mage/node_classification/utils/extract_from_database.py

```diff
@@ -0,0 +1,258 @@
+import torch
+from torch_geometric.data import Data, HeteroData
+import mgp
+import numpy as np
+from tqdm import tqdm
+import random
+from collections import Counter
+import torch_geometric.transforms as T
+import typing
+from collections import defaultdict
+
+
+def nodes_fetching(
+    ctx: mgp.ProcCtx, features_name: str, class_name: str, data: HeteroData
+) -> typing.Tuple[HeteroData, typing.Dict, typing.Dict, str]:
+    """
+    This procedure fetches the nodes from the database and returns them in HeteroData object.
+
+    Args:
+        ctx: The context of the procedure.
+        features_name: The name of the features field.
+        class_name: The name of the class field.
+        data: HeteroData object where nodes will be stored.
+
+    Returns:
+        Tuple of (HeteroData, reindexing, inv_reindexing, observed_attribute).
+    """
+
+    nodes = list(iter(ctx.graph.vertices))  # obtain nodes from context
+    node_types = []  # variable for storing node types
+    embedding_lengths = defaultdict(int)  # variable for storing embedding lengths
+    observed_attribute = None  # variable for storing observed attribute
+
+    for node in nodes:
+
+        if features_name not in node.properties:
+            continue  # if features are not available, skip the node
+
+        # add node type to list of node types (concatenate them if there are more than 1 label)
+        if len(node.labels) > 0:
+            
+            node_type = "_".join(node.labels[i].name for i in range(len(node.labels)))
+            node_types.append(node_type)
+        else:
+            raise Exception(f"Node {node.id} has no labels.")
```

Comment:
```suggestion
        if len(node.labels) == 0:
            raise Exception(f"Node {node.id} has no labels.")
        node_type = "_".join(node.labels[i].name for i in range(len(node.labels)))
        node_types.append(node_type)
```
Less indents and we throw errors before so the code is more clear